crates/driver/src/infra/config/file/mod.rs

```diff
@@ -301,8 +301,11 @@ struct LiquidityConfig {
     #[serde(default)]
     zeroex: Option<ZeroExConfig>,
 
-    /// The base URL used to connect to subgraph clients.
-    graph_api_base_url: Option<Url>,
+    /// The URL used to connect to uniswap v3 subgraph client.
+    uniswap_v3_graph_url: Option<Url>,
```

Comment:
Having the URLs as part of the global config in the `shared` crate is perfectly acceptable because we use CLI parameters to configure the binary which doesn't give us a lot of options in terms of enforcing correctness.

However, in the `driver` crate these arguments should be grouped together with the other parameters for the specific liquidity source. Specifically that means `uniswap_v3_graph_url` should be part of [`UniswapV3Config::Manual`](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/config/file/mod.rs#L371-L390). And since it's already part of the right struct it could be shortened to `graph_url`.
And for `UniswapV3Config::Preset` the user should not have to specify that URL so if somebody uses the `preset` config we'd have to use the default graph URL instead.

Same argument applies for `balancer_v2_graph_url` and `BalancerV2Config::Manual`.

That way a users cannot get surprised that they don't actually index any balancer liquidity although they set the `balancer_v2_graph_url` in the config. Additionally this would allow us to fetch some uni v3 data from one URL and other data from other URLs (but that is not that important).

Reply:
@fredmarques why did you resolve this comment? It's the thing that is blocking the completion of this PR. Currently `uniswap_v3_graph_url` is on the wrong struct (it needs to be renamed and moved into the `UniswapV3Config` struct)